OpenRA.Mods.Common/FileFormats/InstallShieldCABCompression.cs

```diff
@@ -105,41 +109,71 @@ public DirectoryDescriptor(Stream stream, long nameTableOffset)
 
 			public readonly byte[] MD5;
 			public readonly uint NameOffset;
-			public readonly ushort DirectoryIndex;
+			public readonly uint DirectoryIndex;
 			public readonly uint LinkToPrevious;
 
 			public readonly uint LinkToNext;
 			public readonly LinkFlags LinkFlags;
 			public readonly ushort Volume;
 			public readonly string Filename;
 
-			public FileDescriptor(Stream stream, uint index, long tableOffset)
+			public FileDescriptor(Stream stream, uint index, long tableOffset, uint version)
 			{
 				Index = index;
-				Flags = (CABFlags)stream.ReadUInt16();
-				ExpandedSize = stream.ReadUInt32();
-				stream.Position += 4;
-				CompressedSize = stream.ReadUInt32();
-
-				stream.Position += 4;
-				DataOffset = stream.ReadUInt32();
-				stream.Position += 4;
-				MD5 = stream.ReadBytes(16);
 
-				stream.Position += 16;
-				NameOffset = stream.ReadUInt32();
-				DirectoryIndex = stream.ReadUInt16();
-				stream.Position += 12;
-				LinkToPrevious = stream.ReadUInt32();
-				LinkToNext = stream.ReadUInt32();
-
-				LinkFlags = (LinkFlags)stream.ReadUInt8();
-				Volume = stream.ReadUInt16();
-
-				var pos = stream.Position;
-				stream.Position = tableOffset + NameOffset;
-				Filename = stream.ReadASCIIZ();
-				stream.Position = pos;
+				if (version <= 5)
```

Comment:
I assume this condition of version <= 5 would run less often than version > 5, right? If so then it would be better to put the version > 5 case first and the version <= 5 case in the else branch.

Reply:
Hm... the idea was basicaly to have them in ascending order. I took the logic from UnShield as stated before, but it may be possible there are other differences in version 0-4, or above 5. So in my head in the future (while unlikely that this gets changed much at all) might look like
`if (version == 1) ... else if (version == 2) ... else if (version == 2) ... else if (version == 2) ... `